bin/src/learner/raytune_learner.py

```diff
@@ -0,0 +1,243 @@
+        """
+        Initialize the TuneWrapper with the paths to the config, model, experiment and data.
+        """
+        # Load the config
+        self.config = {}
+        with open(config_path, "r") as f:
+            # TODO figure out a better way to load the config
+            self.config =eval(f.read())
+        
+        self.config["model"] = model_path
+        self.config["experiment"] = experiment_path
+
+        try :
+            assert os.path.exists(data_path)
+            self.config["data_path"] = data_path
+        except AssertionError:
+
+            raise ValueError("Data path does not exist. Given path:" + data_path)           
+
+        try:
+            self.scheduler = getattr(schedulers, self.config["scheduler"]["name"])( **self.config["scheduler"]["params"])
+        except AttributeError:
+            raise ValueError(f"Invalid optimizer: {self.config['scheduler']['name']}, check PyTorch for documentation on available optimizers")
+
+        # Get the tune config
+        try: 
+            self.tune_config = getattr(tune, self.config["tune_config"]["name"])(scheduler = self.scheduler, **self.config["tune_config"]["params"])
+        except AttributeError:
+            raise ValueError(f"Invalid tune_config: {self.config['tune_config']['name']}, check PyTorch for documentation on how a tune_config should be defined")
+
+        # Get checkpoint config 
+        try:
+            # TODO: once the file reading is fixed, add the checkpoint_at_end parameter in the config file directly
+            self.checkpoint_config = getattr(train, self.config["checkpoint_config"]["name"])(checkpoint_at_end = False, **self.config["checkpoint_config"]["params"])
+        except AttributeError:
+            raise ValueError(f"Invalid checkpoint_config: {self.config['checkpoint_config']['name']}, check PyTorch for documentation on how a checkpoint_config should be defined")
+
+        # Get the run config
+        try:
+            self.run_config = getattr(train, self.config["run_config"]["name"])(checkpoint_config = self.checkpoint_config, **self.config["run_config"]["params"])
+        except AttributeError:
+            raise ValueError(f"Invalid run_config: {self.config['run_config']['name']}, check PyTorch for documentation on how a run_config should be defined")
+
+        # substite in the checkpoint_config everything that is a bool in string with the actual bool
+        self.tuner = None
+        self.best_config = None
+        self.results = None
+        self.trainer = None
+
+    def _prep_tuner(self):
+        """
+        Prepare the tuner with the configs.
+        """
+        self.tuner = tune.Tuner(TuneModel,
+                            tune_config= self.tune_config,
+                            param_space=self.config,
+                            run_config=self.run_config,
+                        )
+        self.tuner.tuning = True
+
+    def tune(self, overwrite=False):
```

Comment:
Is this an override of the imported tune function from torch? if not is better is not spelled identically, i know is nitpicky but better to be safe.

Reply:
No no, it is the une function of the tune wrapper - so no overwriting :)